src/Microsoft.AspNetCore.Mvc.Core/Internal/ControllerActionInvoker.cs

```diff
@@ -672,12 +690,15 @@ private Task Next(ref State next, ref Scope scope, ref object state, ref bool is
 
                         if (exceptionContext != null)
                         {
-                            if (exceptionContext.Result != null && !exceptionContext.ExceptionHandled)
+                            if (exceptionContext.Result != null ||
+                                exceptionContext.Exception == null ||
+                                exceptionContext.ExceptionHandled)
```

Comment:
Discussed in person and I will add a test that verifies that we **don't** short circuit when you set result. IE any subsequent exception filters still have a chance to run.